docs/src/tutorials/discrete_stochastic_example.md

```diff
@@ -443,8 +577,64 @@ One such algorithm is `SimpleTauLeaping`, which we use as follows:
 sol = solve(jump_prob,SimpleTauLeaping();dt=1.0)
 ```
 
-## Updating `JumpProblem`s
+## FAQ
+
+#### *1. When running many consecutive simulations, for example within an `EnsembleProblem` or loop, how can I update `JumpProblem`s?*
+
 In [Remaking `JumpProblem`s](@ref) we show how to modify parameters, the initial
 condition, and other components of a generated `JumpProblem`. This can be useful
 when trying to call `solve` many times while avoiding reallocations of the
 internal aggregators for each new parameter value or initial condition.
+
+#### *2. How do I use callbacks with `ConstantRateJump` or `MassActionJump` systems?*
+
+Callbacks can be used with `ConstantRateJump`s and `MassActionJump`s. When solving a pure jump system with `SSAStepper`, only discrete callbacks can be used (otherwise a different time stepper is needed). 
+
+*Note, when modifying `u` or `p` within a callback, you must call `reset_aggregated_jumps!(integrator)` after making updates.* This ensures that the underlying jump simulation algorithms know to reinitialize their internal data structures. Leaving out this call will lead to incorrect behavior!
+
+A simple example that uses a `MassActionJump` and changes the parameters at a specified time in the simulation using a `DiscreteCallback` is
+```julia
+using DiffEqJump
+rs = [[1 => 1],[2=>1]]
+ns = [[1 => -1, 2 => 1],[1=>1,2=>-1]]
+p  = [1.0,0.0]
+maj = MassActionJump(rs, ns; param_idxs=[1,2])
+u₀ = [100,0]
+tspan = (0.0,40.0)
+dprob = DiscreteProblem(u₀,tspan,p)
+jprob = JumpProblem(dprob,Direct(),maj)
+pcondit(u,t,integrator) = t==20.0
+function paffect!(integrator)
+  integrator.p[1] = 0.0
+  integrator.p[2] = 1.0
+  reset_aggregated_jumps!(integrator)
+end
+sol = solve(jprob, SSAStepper(), tstops=[20.0], callback=DiscreteCallback(pcondit,paffect!))
+```
+Here at time `20.0` we turn off production of `u[2]` while activating production of `u[1]`, giving
+
+![callback_gillespie](../assets/callback_gillespie.png)
+
+#### *3. How can I define collections of many different jumps and pass them to `JumpProblem`?*
+
+We can use `JumpSet`s to collect jumps together, and then pass them into
+`JumpProblem`s directly. For example, using the `MassActionJump` and
+`ConstantRateJump` defined earlier we can write
+```
+jset = JumpSet(mass_act_jump, birth_jump)
+jump_prob = JumpProblem(prob, Direct(), jset)
+sol = solve(jump_prob, SSAStepper())
+```
+
+#### *4. How can I set the random number generator used in the jump process sampling algorithms (SSAs)?*
+
+Random number generators can be passed to `JumpProblem` via the `rng` keyword
+argument. Continuing the previous example:
+
+```julia
+#] add RandomNumbers
+using RandomNumbers
+jprob = JumpProblem(dprob, Direct(), maj, rng=Xorshifts.Xoroshiro128Star(rand(UInt64)))
+```
+uses the `Xoroshiro128Star` generator from
+[RandomNumbers.jl](https://github.com/JuliaRandom/RandomNumbers.jl).
```

Comment:
This may even make sense to repeat this in the Catalyst docs.

Reply:
Or we can at least link to it from there.

Reply:
that's a better idea.